From 5d39a1a40dc375261aec0a87d1945a17c8031364 Mon Sep 17 00:00:00 2001
From: Subbaraya Sundeep <sbhatta@marvell.com>
Date: Wed, 17 Feb 2021 14:16:34 +0530
Subject: [PATCH 1261/1921] octeontx2-af: Fix VF mac configuration by VF.

PF can set its VF mac address using the
"ip link set dev <PF_eth_interface> vf <VF_id> mac <mac_addr>".
The above command changes the default unicast filter
installed by AF for the VF. Since PF is changing
the AF default unicast rule the command fails because
owner to mcam entry mapping fails. Fix this by relaxing
checks in AF incase PF sets mac for its VF.

Fixes: 878432cb ("octeontx2-af: refactor function npc_install_flow")
Change-Id: I62cb7138c3a692460ed33220ca7ff7306a8e8910
Signed-off-by: Subbaraya Sundeep <sbhatta@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/46176
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/rvu_npc_fs.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu_npc_fs.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu_npc_fs.c
index 9cdd2f04c50c..4978bf0939ef 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu_npc_fs.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu_npc_fs.c
@@ -1046,6 +1046,13 @@ static int npc_install_flow(struct rvu *rvu, int blkaddr, u16 target,
 		rvu_mcam_remove_counter_from_rule(rvu, owner, rule);
 
 	write_req.hdr.pcifunc = owner;
+
+	/* AF owns the default rules so change the owner just to relax
+	 * the checks in rvu_mbox_handler_npc_mcam_write_entry
+	 */
+	if (req->default_rule)
+		write_req.hdr.pcifunc = 0;
+
 	write_req.entry = entry_index;
 	write_req.intf = req->intf;
 	write_req.enable_entry = (u8)enable;
-- 
2.31.1

